cpp/inline/games/stochastic_nim/PerfectPlayer.inl

```diff
@@ -0,0 +1,66 @@
+#include <games/stochastic_nim/PerfectPlayer.hpp>
+
+namespace stochastic_nim {
+
+inline PerfectPlayer::ActionResponse PerfectPlayer::get_action_response(const State& state,
+                                                                 const ActionMask& valid_actions) {
+  if (state.current_mode == kChanceMode) {
+    throw std::invalid_argument("PerfectPlayer does not support chance mode");
+  }
```

Comment:
I have convenience functions to do this slightly more concisely:

```
util::release_assert(state.current_mode == kPlayerMode);
```

Can optionally pass a formatted string with additional format arguments, but here the extra verbosity does not really help code readability or debuggability, so it's not really warranted.